v0/destinations/sendgrid/transform.js

```diff
@@ -0,0 +1,255 @@
+const logger = require("../../../logger");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  getValueFromMessage,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getIntegrationsObj,
+  isEmptyObject
+} = require("../../util");
+const { ENDPOINT } = require("./config");
+const {
+  payloadValidator,
+  eventValidity,
+  createList,
+  createContent,
+  createAttachments,
+  constructFields
+} = require("./util");
+
+const trackResponseBuilder = async (message, { Config }) => {
+  let event = getValueFromMessage(message, "event");
+  if (!event) {
+    throw new CustomError("event is required for track call", 400);
+  }
+  event = event.trim().toLowerCase();
+  eventValidity(Config, event);
```

Comment:
can we return a boolean value for this function and rename it to maybe `isValidEvent`. If false is returned, throw the error from here? Makes the code more readable.